### PR TITLE
Use `create_memref_view_from_dlpack` within Tripy

### PR DESCRIPTION
**Pull request by markkraay**
Relies on #71. Simplifies our Array class by using `create_memref_view_from_dlpack`.

### CODE REVIEW

File: tripy/tests/integration/test_functional.py [hunk 1]
```diff
@@ -166,11 +166,11 @@ def test_weights_loading_from_torch(self, kind):
 
             tripy_linear = tp.Linear(2, 3)
             if kind == "gpu":
-                tripy_linear.weight = tp.Parameter(tp.Tensor(torch_linear.weight.to("cuda"), device=tp.device(kind)))
-                tripy_linear.bias = tp.Parameter(tp.Tensor(torch_linear.bias.to("cuda"), device=tp.device(kind)))
+                tripy_linear.weight = tp.Parameter(tp.Tensor(torch_linear.weight.detach().to("cuda"), device=tp.device(kind)))
```
Comment (jhalakpatel) [inline tripy/tests/integration/test_functional.py:169]: What happens when you remove `detach()`?

Comment (markkraay) [reply tripy/tests/integration/test_functional.py:169]: Torch tensors will not allow invocation of the `__dlpack__` method if they have a `grad` tensor attached.

Comment (jhalakpatel) [reply tripy/tests/integration/test_functional.py:169]: can we instead handle this internally? This could lead to bad UX.

Comment (markkraay) [reply tripy/tests/integration/test_functional.py:169]: Are you suggesting we add a check within the `Array._memref` function to see if the `data`  is a torch tensor and then manually calling `detach()`?

File: tripy/tripy/common/array.py [hunk 2]
```diff
@@ -273,70 +273,7 @@ def _memref(self, data):
...
-                        dtype=convert_tripy_dtype_to_runtime_dtype(self.dtype),
-                    )
-                else:
-                    return self.runtime_client.create_device_memref_view(
-                        ptr,
-                        shape=list(self.shape),
-                        dtype=convert_tripy_dtype_to_runtime_dtype(self.dtype),
-                        device=self.runtime_client.get_devices()[self.device.index],
-                    )
+                return self.runtime_client.create_memref_view_from_dlpack(data.__dlpack__())
```
Comment (jhalakpatel) [inline tripy/tripy/common/array.py:275]: We can only create a host memref view from the host array pointer and a device memref view from the device pointer.
I think we should pass `mlirtrt_device` to this method and check if the pointer type matches the expected device, if we not should throw an error. You use something similar to https://github.com/NVIDIA/TensorRT-Incubator/pull/73/files#diff-1768ba4123934bb7c78a1529f96460e3c2926a3228c036f1a8ffcdd3c982ab20R736 in your MLIR-TRT PR.

Comment (markkraay) [reply tripy/tripy/common/array.py:275]: From my understanding, the DLPack device would correspond to the device that the `data` pointer is on and therefore the device is already determined. Since the device is already determined, passing `mlirtrt_device` and performing an extra check seems unnecessary. Maybe I am misunderstanding something?

Comment (jhalakpatel) [reply tripy/tripy/common/array.py:275]: I agree. I was talking about a mismatch between user set device and location of the data pointer. I can take care of this in a separate PR.

Comment (jhalakpatel) [reply tripy/tripy/common/array.py:275]: Feel free to resolve the thread.

File: tripy/tests/integration/test_functional.py (hunk 1 above)
Comment (jhalakpatel) [reply tripy/tests/integration/test_functional.py:169]: Yeah, that would be better.

File: tripy/tripy/common/array.py (hunk 2 above)
Comment (markkraay) [reply tripy/tripy/common/array.py:275]: Okay I understand now. Will close this.

**Comment by yizhuoz004** [comment]
I'm closing this PR because #95 covers all the changes. Thanks.
